### ISSUE TITLE
Editor crash with player still running.

### ISSUE THREAD

**Issue by engineer-pearl** [bitwes/Gut#433]
Using Steam's Godot, I built a test that crashes Godot with the test window open.
This probably happened because I did not stub my class correctly, however it does not seem right that this should crash the editor. I have included the test file and the file being tested. (extensions changed to make Github happy)

[GameWorld .txt](https://github.com/bitwes/Gut/files/9716955/GameWorld.txt)
[test_GameWorld .txt](https://github.com/bitwes/Gut/files/9716973/test_GameWorld.txt)

Expected outcomes:
- Error messages
- Player crashes, but editor is still up
- Runtime messages

(Basically anything that doesn't crash the editor)

System specs:
OS: Linux Mint
Godot Version: 3.5.1
Language: GDScript

**Comment by bitwes** [comment]
TL;DR I'm pretty sure you are doing something that Godot is not very happy about and decides to just give up.  I do not believe this is a GUT issue.

I was able to replicate the issue with Godot 3.5.1.  When running the test through the editor Godot would sometimes crash.  I used the following test script (basically yours but without reference to the `GameWorld` since it wasn't really being used).  Most of the time I could see the same error from the script below in the Godot Debugger `SCRIPT ERROR: Invalid set index 'AreaMap' (on base: 'Nil') with value of type 'StreamTexture ()'.`
``` gdscript
extends GutTest

var testObject
var doubledStreamTexture

func before_all():
	doubledStreamTexture =  double(StreamTexture).new()
	stub(doubledStreamTexture,'get_pixel').to_return(Vector2(0,0))
	testObject.AreaMap = doubledStreamTexture
	gut.p("ran run setup", 2)

func test_mapChunks_next_to_eachother_on_the_x_share_first_depth_nodes():
	var obj1 = testObject.getMapChunk(0,0,4)
	var obj2 = testObject.getMapChunk(1,0,4)
	assert_call_count(doubledStreamTexture, "get_pixel", 2, [3,1])

func test_mapChunks_next_to_eachother_on_the_y_share_second_depth_nodes():
	var obj1 = testObject.getMapChunk(1,1,4)
	var obj2 = testObject.getMapChunk(1,2,4)
	assert_call_count(doubledStreamTexture, "get_pixel", 2, [4,6])
```

I ran the tests from the command line (without the `-d` debug flag), the output is below.  I believe the `SCRIPT ERROR` entries are what are causing Godot to crash.  I think this is an issue with Godot and not Gut but I'm not completely sure yet.
```
res://test/unit/test_bugs/test_i433.gd
SCRIPT ERROR: Invalid set index 'AreaMap' (on base: 'Nil') with value of type 'StreamTexture ()'.
          at: before_all (res://test/unit/test_bugs/test_i433.gd:13)
* test_mapChunks_next_to_eachother_on_the_x_share_first_depth_nodes
    ran setup
SCRIPT ERROR: Invalid call. Nonexistent function 'getMapChunk' in base 'Nil'.
          at: test_mapChunks_next_to_eachother_on_the_x_share_first_depth_nodes (res://test/unit/test_bugs/test_i433.gd:28)
    ran teardown
    [WARNING]:  Test did not assert
* test_mapChunks_next_to_eachother_on_the_y_share_second_depth_nodes
    ran setup
SCRIPT ERROR: Invalid call. Nonexistent function 'getMapChunk' in base 'Nil'.
          at: test_mapChunks_next_to_eachother_on_the_y_share_second_depth_nodes (res://test/unit/test_bugs/test_i433.gd:33)
    ran teardown
    [WARNING]:  Test did not assert
* after_all
    ran run teardown
```

I was able to get this stack trace when Godot dies:
```
Running: /Applications/Godot.app/Contents/MacOS/Godot --path /Users/butchwesley/development/godot/guts/Gut --remote-debug 127.0.0.1:6007 --allow_focus_steal_pid 46468 --position 0,0 --maximized res://addons/gut/gui/GutRunner.tscn
ERROR: FATAL: Index p_index = 39 is out of bounds (size() = 39).
   at: get (./core/cowdata.h:150)

================================================================
handle_crash: Program crashed with signal 4
Engine version: Godot Engine v3.5.1.stable.official (6fed1ffa313c6760fa88b368ae580378daaef0f0)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] 1   libsystem_platform.dylib            0x00007fff6c8c25fd _sigtramp + 29
[2] 2   ???                                 0x000000000000ffff 0x0 + 65535
[3] SceneTreeEditor::update_tree()
[4] SceneTreeEditor::update_tree()
[5] SceneTreeEditor::update_tree()
[6] Object::get_instance_id() const
[7] Node::get_physics_interpolation_mode() const
[8] Node::get_physics_interpolation_mode() const
[9] DirAccess* DirAccess::_create_builtin<DirAccessOSX>()
[10] DefaultAllocator::alloc(unsigned long)
[11] DirAccess* DirAccess::_create_builtin<DirAccessOSX>()
[12] 12  libdyld.dylib                       0x00007fff6c6c9cc9 start + 1
[13] 13  ???                                 0x0000000000000004 0x0 + 4
-- END OF BACKTRACE --
================================================================
[1]    46468 abort      "/Applications/Godot.app/Contents/MacOS/Godot" --path  -e
```

**Comment by bitwes** [comment]
Reopen if the last comment did not address the issue.
